### ISSUE TITLE
Game Engine in C++ Playing with the Computer

### ISSUE THREAD

**Issue by edogdu** [edogdu/RAM_Cyber_Defenders#21]

**Comment by Luk-Er4** [comment]
On the previous code, I could only play as player vs player game. Since I got the request to make it enable to play as a player vs computer, I had to update the logic. 

Here is the comparison of the blueCard function for example. The previous version is the upper one.

![image](https://github.com/edogdu/RAM_Cyber_Defenders/assets/102909145/c44369b2-66e7-4e0a-a589-b096dea7947e)

![image](https://github.com/edogdu/RAM_Cyber_Defenders/assets/102909145/0cd3e262-7461-448a-82fc-d76bfddb4351)

There is no big difference between the two of them. What I could do for the update was to create a vector to store which position options are available for the AI. Then, I created a new function called "AiMakingChoice" function.

AiMakingChoice function

![image](https://github.com/edogdu/RAM_Cyber_Defenders/assets/102909145/9e60e2bc-e703-4056-895a-bb6d0db21712)

I could pass the vector I mentioned earlier to this function, and then randomly choose among the available choices. The function will return one of those choices, and this is how computer can play the game.
